server/apiRoutes/event.js

```diff
@@ -146,53 +138,57 @@ router.get('/myevents', async (req, res, next) => {
   }
 });
 
-//get all events for a specific group
-router.get('/groups/:id/events', async (req, res, next) => {
-  try {
-    const group = req.params.id;
-    const groupEvents = await group.getEvents();
-    res.send(groupEvents);
-  } catch (err) {
-    next(err);
-  }
-});
+
 
 //add user to attend event
-router.post('/addattendee', async (req, res, next) => {
-  try {
-    let validGroupMember = await GroupMember.findOne({
-      where: { groupId: req.body.groupId, userId: req.user.id },
-    });
-    if (validGroupMember) {
-      await EventAttendee.create({
-        userId: req.user.id,
-        eventId: req.body.eventId,
-      });
-    } else
-      throw new Error(
-        'Events',
-        'You must be a member of this group to attend event'
-      );
-  } catch (err) {
-    next(err);
-  }
+router.post('/addattendee', async(req, res, next) => {
+    try {
+        let validGroupMember = await GroupMember.findOne({ 
+            where: { groupId: req.body.groupId, userId: req.user.id }
+        });
+        if (validGroupMember) {
+            const attendee = await EventAttendee.create({ 
+                userId: req.user.id, eventId: req.body.id 
+            });
+            res.send(attendee)
+        } else throw new Error(
+            'Events', 
+            'You must be a member of this group to attend event'
+        );
+    } catch(err) {
+        next(err);
+    }
 });
 
 //remove user from attending event
 router.delete('/deleteattendee', async (req, res, next) => {
   try {
-    const attendee = await EventAttendee.destroy({
-      where: { userId: req.user.id, eventId: req.body.eventId },
+    const attendee = await EventAttendee.findOne({
+        where: { userId: req.user.id, eventId: req.body.id }
+    })
+    await EventAttendee.destroy({
+      where: { userId: req.user.id, eventId: req.body.id },
+      returning: true
     });
-    if (attendee) res.status(201).send();
-    else throw new Error('Events', 'Unable to delete attendee');
+    res.json(attendee);
   } catch (err) {
     next(err);
   }
 });
 
+//delete event
+//TODO: add restrictions
+
+router.delete('/:id', async (req, res, next) => {
+    try {
+      const deletedEvent = await Event.destroy({ where: { id: req.params.id } }, {returning: true});
+      res.send(deletedEvent[1]);
+    } catch (err) {
+      next(err);
+    }
+  });
```

Comment:
When we delete an event, I think we need to make sure that the EventAttendee model gets updated as well.

Reply:
When I delete the event, it has been deleting from the eventattendee model as well. It looks like it does it automatically